jsapp/js/account/useProducts.hook.ts

```diff
@@ -0,0 +1,37 @@
+import {createContext, useEffect, useState} from 'react';
+import type {Product} from 'js/account/stripe.types';
+import {getProducts} from 'js/account/stripe.api';
+
+export interface ProductsState {
+  products: Product[];
+  isLoaded: boolean;
+}
+
+const INITIAL_PRODUCTS_STATE: ProductsState = Object.freeze({
+  products: [],
+  isLoaded: false,
+});
+
+export function useProducts() {
+  const [products, setProducts] = useState<ProductsState>(
+    INITIAL_PRODUCTS_STATE
+  );
+
+  // get list of products
+  useEffect(() => {
+    getProducts().then((products) => {
+      setProducts(() => {
+        return {
+          products: products.results,
+          isLoaded: true,
+        };
+      });
+    });
+  }, []);
+
+  return products;
```

Comment:
It looks like there's a bug when Stripe isn't enabled. You can replicate by commenting out the `STRIPE_TEST_SECRET_KEY` and `STRIPE_LIVE_MODE` environment variables. This fetch happens even when Stripe is disabled, and 404s. Instead of wrapping this fetch in `useEffect`, you may want to take a look at `useWhenStripeIsEnabled` (which is basically just `useEffect` that only runs once we've confirmed Stripe is active.)

Reply:
@LMNTL Thanks for catching this. I have added the useWhenStripeIsEnabled wrapper as you suggested. 

I also updated the isFullyLoaded memo on the usage page to check for either products being loaded or stripe not being enabled so we don't get an infinite loading spinner there when Stripe is disabled.